### PR TITLE
Added apt-get update for sqlite dependency install

### LINKED ISSUE #19698: 🐛 Bug Report: Build and push Docker image does not work anymore / Images are missing
### 📜 Description

Hello, 
today I wanted to pull version 1.17.5 of backstage and it did not work although the tag is mentioned on the tags list (https://github.com/backstage/backstage/tags). I figured that the "build and push docker image" pipeline does not work anymore. Can you check this?
https://github.com/backstage/backstage/actions/runs/6014499758

### 👍 Expected behavior

I should build all images that are listed in tags: https://github.com/backstage/backstage/tags

### 👎 Actual Behavior with Screenshots

https://github.com/backstage/backstage/actions/runs/6014499758

### 👟 Reproduction steps

docker pull ghcr.io/backstage/backstage:1.17.5

### 📃 Provide the context for the Bug.

I need a newer image. 

### 🖥️ Your Environment

N/A

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Updated Dockerfile and related documentation to include `apt-get update` when installing the SQLite dependency.

Closes #19698 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
This should resolve the errors in this workflow: https://github.com/backstage/backstage/actions/workflows/deploy_docker-image.yml

I can't run this workflow by hand so I'm not sure how we can get a proper image published for the missing versions?

File: docs/tutorials/yarn-migration.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ ENV PYTHON /usr/bin/python3
 # Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
 # in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
+    apt-get update && \
```
Comment (freben) [inline docs/tutorials/yarn-migration.md:96]: So fresh!

**Review by freben — COMMENTED** [review]
Wait. I have a silly question. At what point does docker decide that a RUN layer is stale? :) How might it happen that the layer above (with its apt-get update) won't be immediately followed by the execution of the next RUN line? Because if significant time passed between them, I'd see this fix needed.

File: docs/tutorials/yarn-migration.md (hunk 1 above)
Comment (awanlin) [reply docs/tutorials/yarn-migration.md]: So clean! ;)

**Comment by awanlin** [comment]
> Wait. I have a silly question. At what point does docker decide that a RUN layer is stale? :) How might it happen that the layer above (with its apt-get update) won't be immediately followed by the execution of the next RUN line? Because if significant time passed between them, I'd see this fix needed.

@freben, not a silly question, something I'm still asking myself about this. First, we have a bit of a unique situation here as we want to have the SQLite install on its own so it's clear for those not using it they just remove that entire section. Which means we aren't following the guidance here: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get. Secondly, we don't have this issue when you use `docker image build` but we do when we use `docker buildx build` which is what `docker/build-push-action@v4` uses and that action is being used in the workflow to publish the Backstage container image to the GitHub container registry. 

My thinking with this fix was that it had a fairly low impact and would also fix this for others using the same action or using `buildx` directly.

**Review by freben — APPROVED** [review]
Alright, worst case it gets a little slower, but this is mostly to get people started and then they can tweak as they need (they probably want to add the lists dir deletion too, for example). And correctness trumps some seconds lost

**Comment by awanlin** [comment]
Thanks @freben 🚀

**Pull request merged**